### PR TITLE
Made \Doctrine\ORM\Mapping\Driver\DatabaseDriver bit more adapted for extending

### PR DESCRIPTION
**Pull request by nemutaisama**

### CODE REVIEW

File: lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php [hunk 1]
```diff
@@ -259,24 +255,18 @@ public function loadMetadataForClass($className, ClassMetadata $metadata)
      *
      * @throws \Doctrine\ORM\Mapping\MappingException
      */
-    private function reverseEngineerMappingFromDatabase()
+    protected function reverseEngineerMappingFromDatabase()
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/Driver/DatabaseDriver.php]: Please don't change visibility of the methods: if this class isn't designed for extensibility, then don't extend from it

**Comment by Ocramius** [comment]
Closing as "won't fix". Extending this class is not intended/supported.

**Comment by Nemutaisama** [comment]
@Ocramius sorry, but can you maybe explain me if i doing something wrong?
In my project i want to have bidirectional OneToMany associations for some fields reverse engineered from db. Only way i see to do that is adding some methods to DatabaseDriver, and as for now i need to make copy of Doctrine class, modify it and use it in orm:convert-mapping command instead of default one. Did i missed something?
Thank you.

**Comment by Ocramius** [comment]
> for some fields reverse engineered from db.

You are supposed to reverse engineer once, then add the rest by hand ;-)

**Comment by Nemutaisama** [comment]
Yea, i understand that this is best and probably the only right way. But in my case we don't have complete db structure. So building database architecture going simultaneously with code writing. For this case i see 3 ways:
1. Use table naming template and --filter option when generating Entities - still don't offer 100% guaranty that i won't have situation when rewritten Entity need to be regenerated.
2. Lock finished Entity by chmod (stupid way but it works =)
3. Add to Doctrine functional to use config for add OneToMany relations and Traits using to some Entities. This way can speed up development from database, leaving by hands written code unchanged, but of course at the end manually written code should be merged to Entity file.

And after all - i was just thinking that someone maybe also can find use for this functional =)

**Comment by Ocramius** [comment]
> rewritten Entity need to be regenerated.

That's exactly the misconception: codegen is just for importing legacy DBs.

For all the rest, code the entities (by hand! Yes, it won't kill you!) and use the `orm:validate-schema` command.

**Comment by Nemutaisama** [comment]
Ok, thanks for tip.

**Pull request closed without merging**